workspaces/liferay-adyen-workspace/client-extensions/liferay-adyen-payment-integration/src/main/java/com/liferay/adyen/CaptureRestController.java

```diff
@@ -0,0 +1,102 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.adyen;
+
+import com.adyen.Client;
+import com.adyen.enums.Environment;
+import com.adyen.model.checkout.SessionResultResponse;
+import com.adyen.service.checkout.PaymentsApi;
+
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.json.JSONObject;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * @author Crescenzo Rega
+ */
+@RequestMapping("/capture")
+@RestController
+public class CaptureRestController extends BaseRestController {
+
+	@PostMapping
+	public ResponseEntity<String> post(
+		@AuthenticationPrincipal Jwt jwt, @RequestBody String json) {
+
+		log(jwt, _log, json);
+
+		String errorMessages = null;
+		String paymentStatus = "4";
+
+		try {
+			JSONObject jsonObject = new JSONObject(json);
+
+			JSONObject typeSettingsJSONObject = jsonObject.getJSONObject(
+				"typeSettings");
+
+			JSONObject commercePaymentEntryJSONObject =
+				jsonObject.getJSONObject("commercePaymentEntry");
+
+			SessionResultResponse sessionResultResponse = new PaymentsApi(
+				new Client(
+					typeSettingsJSONObject.getString("apiKey"),
+					Environment.valueOf(
+						typeSettingsJSONObject.getString("environment")))
+			).getResultOfPaymentSession(
+				commercePaymentEntryJSONObject.getString("transactionCode"),
+				commercePaymentEntryJSONObject.getString("sessionResult"), null
+			);
+
+			if (SessionResultResponse.StatusEnum.COMPLETED.compareTo(
+					sessionResultResponse.getStatus()) == 0) {
+
+				paymentStatus = null;
+			}
+			else if (SessionResultResponse.StatusEnum.CANCELED.compareTo(
+						sessionResultResponse.getStatus()) == 0) {
+
+				deleteAdyenWebhooks(
+					"Bearer " + jwt.getTokenValue(),
+					new JSONObject(
+						commercePaymentEntryJSONObject.getString("payload")
+					).getString(
+						"id"
+					),
+					_log);
+
+				paymentStatus = "8";
+			}
+		}
+		catch (Exception exception) {
+			errorMessages = ExceptionUtils.getStackTrace(exception);
```

Comment:
don't like this error handling, what if the integration itself throws an error like the sessionResultResponse contains errors, we are not  using/sending them back

Reply:
this is for all methods not only the capture